Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -355,6 +355,125 @@ void RewriteGlobalVariables::visitEntryPoint(AST::Function& function, AST::Stage
+        UNUSED_PARAM(matrix);
+        return BufferBindingLayout {
+            .type = BufferBindingType::Uniform,
+            .hasDynamicOffset = false,
+            .minBindingSize = 0
+        };
+    }, [&](const Array& array) -> BindGroupLayoutEntry::BindingMember {
+        UNUSED_PARAM(array);
+        return BufferBindingLayout {
+            .type = BufferBindingType::Uniform,
+            .hasDynamicOffset = false,
+            .minBindingSize = 0
+        };
+    }, [&](const Struct& structure) -> BindGroupLayoutEntry::BindingMember {
+        UNUSED_PARAM(structure);
+        return BufferBindingLayout {
+            .type = BufferBindingType::Uniform,
+            .hasDynamicOffset = false,
+            .minBindingSize = 0
+        };
+    }, [&](const Texture& texture) -> BindGroupLayoutEntry::BindingMember {
+        TextureViewDimension viewDimension;
+        bool multisampled = false;
+        bool isStorageTexture = false;
+        switch (texture.kind) {
+        case Types::Texture::Kind::Texture1d:
+            viewDimension = TextureViewDimension::OneDimensional;
+            break;
+        case Types::Texture::Kind::Texture2d:
+            viewDimension = TextureViewDimension::TwoDimensional;
+            break;
+        case Types::Texture::Kind::Texture2dArray:
+            viewDimension = TextureViewDimension::TwoDimensionalArray;
+            break;
+        case Types::Texture::Kind::Texture3d:
+            viewDimension = TextureViewDimension::ThreeDimensional;
+            break;
+        case Types::Texture::Kind::TextureCube:
+            viewDimension = TextureViewDimension::Cube;
+            break;
+        case Types::Texture::Kind::TextureCubeArray:
+            viewDimension = TextureViewDimension::CubeArray;
+            break;
+        case Types::Texture::Kind::TextureMultisampled2d:
+            viewDimension = TextureViewDimension::TwoDimensional;
+            multisampled = true;
+            break;
+
+        case Types::Texture::Kind::TextureStorage1d:
+            isStorageTexture = true;
+            viewDimension = TextureViewDimension::OneDimensional;
+            break;
+        case Types::Texture::Kind::TextureStorage2d:
+            isStorageTexture = true;
+            viewDimension = TextureViewDimension::TwoDimensional;
+            break;
+        case Types::Texture::Kind::TextureStorage2dArray:
+            isStorageTexture = true;
+            viewDimension = TextureViewDimension::TwoDimensionalArray;
+            break;
+        case Types::Texture::Kind::TextureStorage3d:
+            isStorageTexture = true;
+            viewDimension = TextureViewDimension::ThreeDimensional;
+            break;
+        }
+
+        if (isStorageTexture) {
+            return StorageTextureBindingLayout {
+                .viewDimension = viewDimension
+            };
+        }
+
+        return TextureBindingLayout {
+            .sampleType = TextureSampleType::Float,
```

Comment:
This doesn't seem right either

Reply:
`sampleType` is ignored for Metal, so we can write anything which is not `WGPUTextureSampleType_Undefined` but in WGSL this really means we can write any value.

Reply:
Still seems better not to bake this dependency in here, when we can report accurate information easily.

Reply:
I don't believe it is populated by the WGSL compiler and it seems like wasted effort to do so if Metal doesn't use it.

Reply:
Isn’t that the textureType and not the sampleType?